etc/dbus-serialbattery/bms/litime.py

```diff
@@ -0,0 +1,550 @@
+
+        Use_Reason = ""
+        #if last update was long ago we use the current reported by the bms despite it beeing unstable, we also use the current from the BMS if there is a very large discrepency betwen them
+        if time_since_last_update > 120:
+            self.current = Last_few_avg
+            Use_Reason = "curr: over 120s since last remaining_amph update"
+
+        elif self.Last_remianAh_initiation != 2:
+            self.current = Last_few_avg
+            Use_Reason = "curr: Last_remianAh not initiated with base values"
+
+        elif time_since_last_update > 5 and (self.current_based_on_remaning + 3 < self.current or self.current_based_on_remaning - 3 > self.current):
+            self.current = Last_few_avg
+            Use_Reason = "curr: Large differances betwen base and curr despite recent base update"
+
+        else:
+            self.current = self.current_based_on_remaning
+            Use_Reason = "base"
+
+        logger.info(f"{Use_Reason}, current:{current:.3f}, Last_few_avg: {Last_few_avg:.3f}, base: {self.current_based_on_remaning:.3f}")#, cur1: {cur1}, cur2: {cur2}, c1: {c1}, c2: {c2}, c3: {c3}, c4: {c4}")
+
+        #Overwrite earlier logic
+        #self.current = current
+
+        # temperature sensor 1 in °C (float)
+        temp1 = cell_temp
+        self.to_temp(1, temp1)
+
+        # status of the battery if charging is enabled (bool)
+        self.charge_fet = True
+        if battery_state == 4:
+            self.charge_fet = False
+
+        # status of the battery if discharging is enabled (bool)
+        self.discharge_fet = True
+
+        self.capacity_remaining = remaining_amph
+
+        # temperature sensor 2 in °C (float)
+        temp2 = unknown_temp
+        self.to_temp(2, temp2)
+
+        # temperature sensor MOSFET in °C (float)
+        temp_mos = mosfet_temp
+        self.to_temp(0, temp_mos)
+
+        self.history.total_ah_drawn = discharges_amph_count
+
+        self.history.full_discharges = discharges_count
+
+    def get_settings(self):
+        """
+        After successful connection get_settings() will be called to set up the battery
+        Set all values that only need to be set once
+        Return True if success, False for failure
+        """
+        logger.info("get_settings")
+
+
+        # MANDATORY values to set
+        # does not need to be in this function, but has to be set at least once
+        # could also be read in a function that is called from refresh_data()
+        #
+        # if not available from battery, then add a section in the `config.default.ini`
+        # under ; --------- BMS specific settings ---------
+
+        # number of connected cells (int)
+        #self.cell_count = 8 #VALUE_FROM_BMS
+
+        # capacity of the battery in ampere hours (float)
+        #self.capacity = 100 #VALUE_FROM_BMS
+
+        # init the cell array once XXXXX FIX 8 hardcoded
+        for _ in range(8):
```

Comment:
From what I saw from their website it‘s possible that a battery has 4, 8 or 16 cells. Is this only temporary?